instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py

```diff
@@ -646,3 +682,34 @@ def _get_runtime_gc_count(
         for index, count in enumerate(gc.get_count()):
             self._runtime_gc_count_labels["count"] = str(index)
             yield Observation(count, self._runtime_gc_count_labels.copy())
+
+    def _get_runtime_thread_count(
+        self, options: CallbackOptions
+    ) -> Iterable[Observation]:
+        """Observer callback for runtime active thread count"""
+        yield Observation(
+            self._proc.num_threads(), self._runtime_thread_count_labels.copy()
+        )
+
+    def _get_runtime_cpu_utilization(
+        self, options: CallbackOptions
+    ) -> Iterable[Observation]:
+        """Observer callback for runtime CPU utilization"""
+        proc_cpu_percent = self._proc.cpu_percent()
```

Comment:
Looking at the [documentation](https://psutil.readthedocs.io/en/latest/index.html?highlight=cpu_percent#psutil.Process.cpu_percent) for self._proc.cpu_percent(), it seems like the first call will return a meaningless 0.0 value?

Reply:
The provided code is a part of the cpu_percent method from the Process class in psutil. It calculates the CPU percent using the delta from cpu_times. If the instance doesn't have _last_proc_cpu_times, it returns 0.0. This means that it will return 0.0 on its first call. To address this issue, one could run self._proc.cpu_percent() once beforehand. By doing so, the meaningless 0.0 data won't be returned. However, this approach might make the code look a bit odd.

```python
        else:
            st1 = self._last_sys_cpu_times
            pt1 = self._last_proc_cpu_times
            st2 = timer()
            pt2 = self._proc.cpu_times()
            if st1 is None or pt1 is None:
                self._last_sys_cpu_times = st2
                self._last_proc_cpu_times = pt2
                return 0.0

        delta_proc = (pt2.user - pt1.user) + (pt2.system - pt1.system)
        delta_time = st2 - st1
        # reset values for next call in case of interval == None
        self._last_sys_cpu_times = st2
        self._last_proc_cpu_times = pt2
```

Reply:
Perhaps returning one instance of 0.0 for the first metric is fine.